rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,528 @@
+            )
+            return final_data_no_pag
+
+        # If there is pagination we follow the example from the official ftx python example
+        # https://github.com/ftexchange/ftx/blob/master/rest/client.py#L163
+        # In this case the strategy is a while loop leaving fixed the start_time (lower bound)
+        # and decreasing end time (the upper bound) until we fetch all the available information
+        new_end_time = end_time
+        ids = set()
+        final_data: List[Dict[str, Any]] = []
+        while True:
+            step = self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=new_end_time,
+            )
+
+            assert isinstance(step, list), 'Doing pagination in something that is not a list'
+
+            # remove possible duplicates
+            deduped = [
+                r for r in step if
+                'id' in r.keys() and r['id'] not in ids
+            ]
+            ids |= {r['id'] for r in deduped}
+            final_data.extend(deduped)
+
+            if len(step) == 0:
+                break
+
+            # Avoid deserialization error if there is a bad date
+            times = []
+            for t in step:
+                try:
+                    times.append(deserialize_timestamp_from_date(t['time'], 'iso8601', 'ftx'))
+                except (DeserializationError, KeyError):
+                    continue
+
+            if len(times) != 0:
+                new_end_time = min(times)
+            else:
+                log.error(
+                    f'Error processing ftx trade history. Fill query step '
+                    f'returned invalid entries when trying pagination for endpoint '
+                    f'{endpoint} with start_time {start_time}, end_time {end_time} '
+                    f'and limit of {limit}',
+                )
+                break
+
+            if len(step) < limit:
+                break
+
+        return final_data
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> ExchangeQueryBalances:
+        try:
+            resp = self._api_query('wallet/all_balances', paginate=False)
+        except RemoteError as e:
+            msg = (
+                f'FTX API request failed. Could not reach FTX due '
+                f'to {e}'
+            )
+            log.error(msg)
+            return None, msg
+
+        # flatten the list that maps accounts to balances
+        balances = [x for _, bal in resp.items() for x in bal]
+
+        # extract the balances and aggregate them
+        returned_balances: DefaultDict[Asset, Balance] = defaultdict(Balance)
+        for balance_info in balances:
+            try:
+                amount = deserialize_asset_amount(balance_info['total'])
+
+                # ignore empty balances. FTX returns zero for some coins
+                # I believe those that the user previously owned
```

Comment:
no need for the second line of the comment. If you verified it put it. If not, no need to have guesses in the comments.